### ISSUE TITLE
[Windows] Laravel CSFR on mapped directory.

### ISSUE THREAD

**Issue by hitech95** [laravel/homestead#1215]
## Versions
  - Vagrant: Vagrant 2.2.5
  - Provider: Virtualbox 6.0
  - Homestead: Latest available today

## Host operating system
I'm running Windows 10 Pro - 64bit

## Homestead.yaml
```yaml
---
ip: "192.168.10.10"
memory: 2048
cpus: 2
provider: virtualbox
mariadb: true

authorize: ~/.ssh/id_rsa.pub

keys:
    - ~/.ssh/id_rsa

folders:
    - map: ~/Documents/GitHub/RSCS
      to: /home/vagrant/rscs

sites:
    - map: laravel.test
      to: /home/vagrant/laravel/public
    - map: rscs.test
      to: /home/vagrant/rscs/public

databases:
    - homestead
```

## Vagrant destroy & up output
My link to [Vagrant UP](https://gist.github.com/hitech95/8c7cc22d3d09dc6e82dc0a006988c4a5) 
## Expected behavior
No problem with the mapped folder files.

## Actual behavior
After creating a  laravel project I've tried to login on the app. (I've created the basic auth).
When you try to login or/and register the page throw a 'csrf token mismatch' error.

The laravel project is inside a mapped directory of my windows machine. (C:Users\hitech95\Documents\GitHub\RSCS)

If I copy that content of that folder directly on the VM (Homestead) in a folder 'laravel' (outside the mapped directory) and using the site 'laravel.test' on my Homestead.yaml it work without problem.

Looks like something wrong with the permissions. If I move the sessions to DB i don't have this problem. I'm not sure if with this workaround I'll find other issues with the filesystem.
The `vagrant up` command has been run on a administrator level shell. (On Windows)

## Steps to reproduce
1. Install Virtualbox
2. Install Vagrant
3. Install the homestead box
4. Clone the Homestead repo
5. Map a working laravel install from the windows host
6. Try to use sessions (exaple login to the laravel installation)

The fact that i can reproduce this only on a mapped folder make me think its a problem related with homestead. If not please tell me.

**Comment by svpernova09** [comment]
Are you running Vagrant from the same user that owns the shared files? Does the user that's running vagrant have permissions to the folder?

Try mapping the full folder path and not using `~` in the folders section.

**Comment by hitech95** [comment]
Hi,
Yes I've tried to run the `vagrant up` with both _hitech95_ and _Administrator_ users.
With the same result. (The project is inside a sub folder of my account documents folder)

What is quite strange is the fact that the session files are created under `/storage/framework/sessions`.

On the session file the token is the same as the one in the form.

I've just tried with the fullpath and it's working.

I suggest to add a note/warning on the docs for Windows users.
